ecc_enclave/enclave/enclave.cpp

```diff
@@ -248,31 +98,105 @@ int ecall_cc_invoke(const uint8_t* signed_proposal_proto_bytes,
         // TODO: create fabric Response object
         // TODO: encrypt fabric Response object
         crm = {};
-        crm.encrypted_response = (pb_bytes_array_t*)pb_realloc(
-            crm.encrypted_response, PB_BYTES_ARRAY_T_ALLOCSIZE(b64_response.length()));
-        COND2LOGERR(crm.encrypted_response == NULL, "cannot allocate encrypted message");
-        crm.encrypted_response->size = b64_response.length();
-        ret = memcpy_s(crm.encrypted_response->bytes, crm.encrypted_response->size,
-            b64_response.c_str(), b64_response.length());
-        COND2LOGERR(ret != 0, "cannot encode field");
 
-        // serialize enclave id
-        enclave_id = g_cc_data->get_enclave_id();
-        crm.enclave_id = (char*)pb_realloc(crm.enclave_id, enclave_id.length() + 1);
-        ret =
-            memcpy_s(crm.enclave_id, enclave_id.length(), enclave_id.c_str(), enclave_id.length());
-        crm.enclave_id[enclave_id.length()] = '\0';
-        COND2LOGERR(ret != 0, "cannot encode enclave id");
+        {  // serialize encrypted response
+            crm.encrypted_response = (pb_bytes_array_t*)pb_realloc(
+                crm.encrypted_response, PB_BYTES_ARRAY_T_ALLOCSIZE(b64_response.length()));
+            COND2LOGERR(crm.encrypted_response == NULL, "cannot allocate encrypted message");
+            crm.encrypted_response->size = b64_response.length();
+            ret = memcpy_s(crm.encrypted_response->bytes, crm.encrypted_response->size,
+                b64_response.c_str(), b64_response.length());
+            COND2LOGERR(ret != 0, "cannot encode field");
+        }
+
+        {  // serialize enclave id
```

Comment:
Isn't the serialize misleading here and for rwset below?  What you create here is first the (non-serialized/in-memory) crm protobuf object and the serialize is really only what you call the encode step?  the proposal is somewhat a special case as you re-use the already serialized version (so the comment "serialize" is also misleading, just for different reasons?)

seems for signature the comments are much clearer (with fill). Probably best to use same terminology (and sequence, i.e., first object creation, then encoding) here as below (with additional explicitly pointing out that for proposal you reuse already serialized version and maybe just moving it right in front the encoding part?)